crates/phactory/src/system/mod.rs

```diff
@@ -730,10 +845,20 @@ impl<Platform: pal::Platform> System<Platform> {
     ) {
         info!("Incoming gatekeeper launch event: {:?}", event);
         match event {
-            GatekeeperLaunch::FirstGatekeeper(new_gatekeeper_event) => {
-                self.process_first_gatekeeper_event(block, origin, new_gatekeeper_event)
+            GatekeeperLaunch::FirstGatekeeper(event) => {
+                if !origin.is_pallet() {
```

Comment:
Now all the places has `origin.is_pallet()`  checked. Consider to move it before `match`?

Reply:
Changed.